pkg/controllers/tlspolicy/tlspolicy_controller.go

```diff
@@ -0,0 +1,269 @@
+		return err
+	}
+
+	// set annotation of policies affecting the gateway - should be the last step, only when all the reconciliation steps succeed
+	return r.ReconcileGatewayPolicyReferences(ctx, tlsPolicy, gatewayDiffObj)
+}
+
+func (r *TLSPolicyReconciler) deleteResources(ctx context.Context, tlsPolicy *v1alpha1.TLSPolicy, targetNetworkObject client.Object) error {
+	// delete based on gateway diffs
+
+	gatewayDiffObj, err := r.ComputeGatewayDiffs(ctx, tlsPolicy, targetNetworkObject, &TLSPolicyRefsConfig{})
+	if err != nil {
+		return err
+	}
+
+	if err := r.reconcileCertificates(ctx, tlsPolicy, gatewayDiffObj); err != nil {
+		return err
+	}
+
+	// remove direct back ref
+	if targetNetworkObject != nil {
+		if err := r.DeleteTargetBackReference(ctx, client.ObjectKeyFromObject(tlsPolicy), targetNetworkObject, TLSPolicyBackRefAnnotation); err != nil {
+			return err
+		}
+	}
+
+	// update annotation of policies affecting the gateway
+	return r.ReconcileGatewayPolicyReferences(ctx, tlsPolicy, gatewayDiffObj)
+}
+
+func (r *TLSPolicyReconciler) calculateStatus(tlsPolicy *v1alpha1.TLSPolicy, specErr error) *v1alpha1.TLSPolicyStatus {
+	newStatus := tlsPolicy.Status.DeepCopy()
+	if specErr != nil {
+		newStatus.ObservedGeneration = tlsPolicy.Generation
+	}
+	readyCond := r.readyCondition(string(tlsPolicy.Spec.TargetRef.Kind), specErr)
+	meta.SetStatusCondition(&newStatus.Conditions, *readyCond)
+	return newStatus
+}
+
+func (r *TLSPolicyReconciler) readyCondition(targetNetworkObjectectKind string, specErr error) *metav1.Condition {
+	cond := &metav1.Condition{
+		Type:    conditions.ConditionTypeReady,
+		Status:  metav1.ConditionTrue,
+		Reason:  fmt.Sprintf("%sTLSEnabled", targetNetworkObjectectKind),
+		Message: fmt.Sprintf("%s is TLS Enabled", targetNetworkObjectectKind),
+	}
+
+	if specErr != nil {
+		cond.Status = metav1.ConditionFalse
+		cond.Reason = "ReconciliationError"
+		cond.Message = specErr.Error()
+	}
+
+	return cond
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *TLSPolicyReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	gatewayEventMapper := events.NewGatewayEventMapper(r.Logger(), &TLSPolicyRefsConfig{}, "tlspolicy")
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&v1alpha1.TLSPolicy{}).
+		Watches(
+			&source.Kind{Type: &gatewayapiv1beta1.Gateway{}},
+			handler.EnqueueRequestsFromMapFunc(gatewayEventMapper.MapToPolicy),
+		).
+		Complete(r)
+}
+
+// The following methods are here temporarily and copied from the kuadrant-operator https://github.com/Kuadrant/kuadrant-operator/blob/main/pkg/reconcilers/targetref_reconciler.go#L45
+// FetchValidTargetRef and FetchValidGateway currently expect that the gateway should have a ready condition, but in the
+// case of the TLSPolicy it might not be ready because there is an invalid tls section that this policy would resolve.
+// ToDo mnairn: Create issue in kuadrant-operator and link it here
```

Comment:
Issue been created yet?
Any reason they can't be referenced from the kuadrant-operator?

Reply:
> Issue been created yet? 

No, will do it now and update.

> Any reason they can't be referenced from the kuadrant-operator?

You mean why is this not using the kuadrant operator functions? Reason is this check https://github.com/Kuadrant/kuadrant-operator/blob/main/pkg/reconcilers/targetref_reconciler.go#L55-L57. In the case of TLS we don't want it to do that since it could be not ready because of TLS and we want this to continue to create the TLS certificate and secrets that might bring the gateway into a programmed state.